libraries/kotlin.test/common/src/main/kotlin/kotlin/test/Assertions.kt

```diff
@@ -72,6 +72,21 @@ fun <@OnlyInputTypes T> assertNotSame(illegal: T, actual: T, message: String? =
     asserter.assertNotSame(message, illegal, actual)
 }
 
+/** Asserts that [value] is [T], with an optional [message]. */
```

Comment:
```suggestion
/** Asserts that [value] is of type [T], with an optional [message]. */
```

Also makes sense to add a note that due to type erasure the type check may be partial, e.g. `assertIs<List<String>>(value)` only checks for the class being `List` and doesn't check the type of elements because it's erased.